lib/api/objectPutCopyPart.js

```diff
@@ -77,60 +77,59 @@ function objectPutCopyPart(authInfo, request, sourceBucket,
     };
 
     return async.waterfall([
-        function checkSourceAuthorization(next) {
+        function checkDestAuth(next) {
+            return services.metadataValidateAuthorization(valPutParams,
```

Comment:
Same here. I would recommend calling `metadata.getBucket` and use the bucket directly in `checkSourceAuthorization` step by bringing the logic directly from `services.metadataValidateAuthorization`.